src/System.Private.ServiceModel/tests/Scenarios/Binding/Custom/CustomBindingTests.cs

```diff
@@ -51,4 +51,69 @@ public static void DefaultSettings_Tcp_Binary_Echo_RoundTrips_String()
             ScenarioTestHelpers.CloseCommunicationObjects((ICommunicationObject)serviceProxy, factory);
         }
     }
+
+    public class MessageModifier : ChannelMessageInterceptor
+    {
```

Comment:
Thanks Ron and Stephen! That makes sense. Let me move this new test to Scenarios/Extensibility